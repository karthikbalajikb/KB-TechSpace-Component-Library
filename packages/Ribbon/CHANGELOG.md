# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.2.0](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/compare/@kb-techspace-org/ribbon@1.1.0...@kb-techspace-org/ribbon@1.2.0) (2019-09-15)

### Features

- **component:** updated package config ([fa43bcb](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/fa43bcb))

# 1.1.0 (2019-09-14)

### Features

- **ribbon:** ribbon component ([a6fc36e](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/commit/a6fc36e)), closes [#1](https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/issues/1)
