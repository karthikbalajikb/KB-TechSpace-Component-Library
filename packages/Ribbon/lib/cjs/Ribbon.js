'use strict';
var __makeTemplateObject =
  (this && this.__makeTemplateObject) ||
  function(cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', { value: raw });
    } else {
      cooked.raw = raw;
    }
    return cooked;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, '__esModule', { value: true });
var react_1 = __importDefault(require('react'));
var styled_components_1 = __importDefault(require('styled-components'));
var Ribbon = function(_a) {
  var title = _a.title;
  return react_1.default.createElement(
    Container,
    null,
    react_1.default.createElement(RibbonText, null, title),
  );
};
exports.default = Ribbon;
var Container = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        "\n  background-color: #8acb82;\n  display: inline-block;\n  color: white;\n  padding: 0em 0.25em 0.25em 0.25em;\n  margin: 27px;\n  position: relative;\n  top: 0.85em;\n  margin-bottom: 20px;\n  margin-top: 20px;\n\n  &:before {\n    content: '';\n    background-color: #8acb82;\n    padding: 0em 0.5em;\n    left: -0.75em;\n    top: -0.2em;\n    bottom: auto;\n    height: 1.35em;\n    position: absolute;\n  }\n\n  &:after {\n    content: '';\n    background-color: #8acb82;\n    padding: 0em 0.5em;\n    left: auto;\n    top: 0.3em;\n    bottom: auto;\n    height: 1.35em;\n    position: absolute;\n  }\n",
      ],
      [
        "\n  background-color: #8acb82;\n  display: inline-block;\n  color: white;\n  padding: 0em 0.25em 0.25em 0.25em;\n  margin: 27px;\n  position: relative;\n  top: 0.85em;\n  margin-bottom: 20px;\n  margin-top: 20px;\n\n  &:before {\n    content: '';\n    background-color: #8acb82;\n    padding: 0em 0.5em;\n    left: -0.75em;\n    top: -0.2em;\n    bottom: auto;\n    height: 1.35em;\n    position: absolute;\n  }\n\n  &:after {\n    content: '';\n    background-color: #8acb82;\n    padding: 0em 0.5em;\n    left: auto;\n    top: 0.3em;\n    bottom: auto;\n    height: 1.35em;\n    position: absolute;\n  }\n",
      ],
    )),
);
var RibbonText = styled_components_1.default.i(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        "\n  padding: 0em 3em;\n  font-size: 11px;\n  text-transform: uppercase;\n  position: relative;\n  display: inline-block;\n  font-weight: normal;\n  letter-spacing: 0.2em;\n  white-space: nowrap;\n  text-align: center;\n\n  &:before {\n    content: '';\n    position: absolute;\n    bottom: auto;\n    top: -0.8em;\n    border-bottom: 0.5em solid #080b0c;\n    border-bottom: 0.5em solid rgba(8, 11, 12, 0.2);\n    border-left: 0.5em solid transparent;\n    left: -0.55em;\n  }\n\n  &:after {\n    content: '';\n    position: absolute;\n    bottom: auto;\n    z-index: 1;\n    top: 1.5em;\n    left: auto;\n    right: -0.5em;\n    pointer-events: none;\n    -webkit-mask: none;\n    border-top: 0.5em solid rgba(8, 11, 12, 0.2);\n    border-right: 0.5em solid transparent;\n  }\n",
      ],
      [
        "\n  padding: 0em 3em;\n  font-size: 11px;\n  text-transform: uppercase;\n  position: relative;\n  display: inline-block;\n  font-weight: normal;\n  letter-spacing: 0.2em;\n  white-space: nowrap;\n  text-align: center;\n\n  &:before {\n    content: '';\n    position: absolute;\n    bottom: auto;\n    top: -0.8em;\n    border-bottom: 0.5em solid #080b0c;\n    border-bottom: 0.5em solid rgba(8, 11, 12, 0.2);\n    border-left: 0.5em solid transparent;\n    left: -0.55em;\n  }\n\n  &:after {\n    content: '';\n    position: absolute;\n    bottom: auto;\n    z-index: 1;\n    top: 1.5em;\n    left: auto;\n    right: -0.5em;\n    pointer-events: none;\n    -webkit-mask: none;\n    border-top: 0.5em solid rgba(8, 11, 12, 0.2);\n    border-right: 0.5em solid transparent;\n  }\n",
      ],
    )),
);
var templateObject_1, templateObject_2;
//# sourceMappingURL=Ribbon.js.map
