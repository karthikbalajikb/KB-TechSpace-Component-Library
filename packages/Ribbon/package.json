{
  "name": "@kb-techspace-org/ribbon",
  "version": "1.0.0",
  "description": "Ribbon Component",
  "author": "Karthik Balaji <karthik@auzmor.com>",
  "homepage": "https://github.com/karthikbalajikb/KB-TechSpace-Component-Library#readme",
  "license": "ISC",
  "main": "lib/Ribbon.js",
  "publishConfig": {
    "access": "public"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/karthikbalajikb/KB-TechSpace-Component-Library.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1",
    "start": "tsc -w"
  },
  "bugs": {
    "url": "https://github.com/karthikbalajikb/KB-TechSpace-Component-Library/issues"
  }
}
