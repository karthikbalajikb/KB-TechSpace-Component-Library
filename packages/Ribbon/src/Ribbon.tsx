import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

interface Props {
  /**
   * text to be displaced in the ribbon
   */
  title: string;
}

const Ribbon: React.FC<Props> = ({ title }) => (
  <Container>
    <RibbonText>{title}</RibbonText>
  </Container>
);

Ribbon.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Ribbon;

const Container = styled.div`
  background-color: #8acb82;
  display: inline-block;
  color: white;
  padding: 0em 0.25em 0.25em 0.25em;
  margin: 27px;
  position: relative;
  top: 0.85em;
  margin-bottom: 20px;
  margin-top: 20px;

  &:before {
    content: '';
    background-color: #8acb82;
    padding: 0em 0.5em;
    left: -0.75em;
    top: -0.2em;
    bottom: auto;
    height: 1.35em;
    position: absolute;
  }

  &:after {
    content: '';
    background-color: #8acb82;
    padding: 0em 0.5em;
    left: auto;
    top: 0.3em;
    bottom: auto;
    height: 1.35em;
    position: absolute;
  }
`;

const RibbonText = styled.i`
  padding: 0em 3em;
  font-size: 11px;
  text-transform: uppercase;
  position: relative;
  display: inline-block;
  font-weight: normal;
  letter-spacing: 0.2em;
  white-space: nowrap;
  text-align: center;

  &:before {
    content: '';
    position: absolute;
    bottom: auto;
    top: -0.8em;
    border-bottom: 0.5em solid #080b0c;
    border-bottom: 0.5em solid rgba(8, 11, 12, 0.2);
    border-left: 0.5em solid transparent;
    left: -0.55em;
  }

  &:after {
    content: '';
    position: absolute;
    bottom: auto;
    z-index: 1;
    top: 1.5em;
    left: auto;
    right: -0.5em;
    pointer-events: none;
    -webkit-mask: none;
    border-top: 0.5em solid rgba(8, 11, 12, 0.2);
    border-right: 0.5em solid transparent;
  }
`;
