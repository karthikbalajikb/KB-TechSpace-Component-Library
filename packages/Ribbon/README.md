# `Ribbon`

> Primary Ribbon component

## Usage

```js
import React, { Component } from 'react';
import Ribbon from '@kb-techspace-org/ribbon';

export default class App extends Component {
  render() {
    return <Ribbon title="hello" />;
  }
}
```
