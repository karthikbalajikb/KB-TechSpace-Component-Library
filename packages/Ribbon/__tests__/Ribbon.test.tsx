import React from 'react';
import { render } from '@testing-library/react';

import Ribbon from '../src/Ribbon';

describe('Ribbon', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Ribbon title="demo" />);
    expect(asFragment().firstChild).toMatchSnapshot();
    // expect(getByText('demo')).toBeInTheDocument();
  });
});
