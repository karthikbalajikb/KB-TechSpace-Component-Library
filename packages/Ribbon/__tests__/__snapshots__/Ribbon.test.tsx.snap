// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon renders correctly 1`] = `
<div
  class="sc-bdVaJa XpfBE"
>
  <i
    class="sc-bwzfXH ibtxQC"
  >
    demo
  </i>
</div>
`;
